NotifyMe/serializers.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from channels.layers import get_channel_layer
+from asgiref.sync import async_to_sync
+from django.db import IntegrityError
+from rest_framework.exceptions import ValidationError
+from rest_framework import serializers
+from .models.notification import Notification 
+from .models.subscription import Subscription
+from .models.subscriptionPlan import SubscriptionPlan
+from .models.user import User
+from django.utils import timezone
+from datetime import timedelta
+from .models.notificationType import NotificationType
+from .constants import *
+
+logger = logging.getLogger(__name__)
+
+class UserSerializer(serializers.ModelSerializer):
+    subscription_plan = serializers.PrimaryKeyRelatedField(queryset=SubscriptionPlan.objects.all())
+
+    class Meta:
+        model = User
+        fields = ('id', 'email_id', 'first_name', 'last_name', 'subscription_plan', 'created_at', 'updated_at')
+        depth = 1
+
+    def get_endtime(self, subscription_plan, start_date):
+        if subscription_plan.subscription_plan == plans["BASIC_PLAN"]:
+            return start_date + timedelta(days=30)
+        elif subscription_plan.subscription_plan == plans["REGULAR_PLAN"]:
+            return start_date + timedelta(days=3 * 30)
+        elif subscription_plan.subscription_plan == plans["STANDARD_PLAN"]:
+            return start_date + timedelta(days=6 * 30)
+        elif subscription_plan.subscription_plan == plans["PREMIUM_PLAN"]:
+            return start_date + timedelta(days=365)
+        else:
+            return None
+
+    def create(self, validated_data):
+        try:
+            subscription_plan = validated_data.pop('subscription_plan')
+
+            if subscription_plan is None:
+                raise ValidationError("The 'subscription_plan' field is required.")
+
+            user = User.objects.create(subscription_plan=subscription_plan, **validated_data)
+
+            # Create a new Subscription
+            start_date = timezone.now()
+            end_date = self.get_endtime(subscription_plan, start_date)
+            if end_date is None:
+                raise ValidationError("Invalid subscription plan")
+
+            Subscription.objects.create(
+                user_id=user,
+                subscription_plan=subscription_plan,
+                start_date=start_date,
+                end_date=end_date    
+            )
+
+            logger.info(f"User created successfully with ID: {user.id}")
+
+            channel_layer = get_channel_layer()
```

Comment:
send notification move to new class called NotificationManager there declare this

Reply:
I have deleted that